pyro/infer/contract.py

```diff
@@ -213,52 +213,6 @@ def broadcast(self, term, ordinal):
         return term
 
 
-def _check_tree_structure(tensor_tree):
-    # Check for absence of diamonds, i.e. collider iaranges.
-    # This is required to avoid loops in message passing.
-    #
-    # For example let f1, f2 be two CondIndepStackFrames, and consider
```

Comment:
This is a nice figure but is no longer correct. I'll try to add something like it to both the tutorial #1355 and the tech report #1391 .